### ISSUE TITLE
Misleading warning "Expected a configuration file"

### ISSUE THREAD

**Issue by paugier** [exabl/snek5000#47]
More a remark/question than a real issue. I'm not a big fan of this warning:

```
WARNING  Expected a configuration file describing compilers and flags:                                             base.py:130
                             /home/pierre/Dev/snek5000-cbox/src/snek5000_cbox/etc/gre.yml                                                         
                    INFO     Using default configuration instead: /home/pierre/Dev/snek5000/src/snek5000/assets/default_configfile.yml base.py:135
```

It is triggered here: https://github.com/exabl/snek5000/blob/master/src/snek5000/output/base.py#L130

The problem is that the warning does not explain how to fix the problem and second, in many cases, there is actually no problem because the default configuration is correct.

I don't know what would be the better thing to do. Maybe just trigger this warning just in debug mode? Maybe also implement local configuration file and mention that in the doc?

**Comment by ashwinvis** [comment]
This is easy to fix. I would like to keep it as a **warning** since the choice of compiler and flags are quite important and should not go unnoticed.

How about we reword it to be helpful
```
WARNING  Missing a configuration file describing compilers and flags. Create a YAML configuration file at the follwoing location to avoid future warnings:                                             base.py:130
                             /home/pierre/Dev/snek5000-cbox/src/snek5000_cbox/etc/gre.yml
INFO     Using default configuration for now: /home/pierre/Dev/snek5000/src/snek5000/assets/default_configfile.yml base.py:135
```

**Comment by ashwinvis** [comment]
To elaborate why I think a warning is needed: gcc / gfortran is hardcoded (for now) the way configuration is loaded. I don't want (future me or anyone) to assume that a `module load intel` would set the compiler.

**Comment by paugier** [comment]
Rewording could be a good option but that's really strange to tell a user to add one file per computer in the package. Wouldn't it be better to tell the user to add a local conf file, something like `~/.config/snek5000.yml`?

**Comment by paugier** [comment]
Another thing, environment variable should be "stronger" than a config file. Setting CC should change the behavior, even with a config file. This is for example Pythran behavior. So `module load intel` should work. Don't you thing so?

**Comment by ashwinvis** [comment]
These are good suggestions. I will work on those.

- [] Local config file at `XDG_CONFIG_HOME`
- [] Environment variable override possibility
